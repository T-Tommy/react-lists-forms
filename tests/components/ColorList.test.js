import React from 'react';
import { shallow } from 'enzyme';
import ColorList from '../../src/components/ColorList';

describe('ColorList component', () => {
  it('renders ColorList', () => {
    const wrapper = shallow(<ColorList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updates the state', () => {
    const color = {
      name: 'Blaci',
      hex: '#424242'
    };

    const wrapper = shallow(<ColorList />);
    wrapper.instance().addColor(color);
    expect(wrapper.state('colors')).toEqual([
      {
        name: 'Blaci',
        hex: '#424242'
      }
    ]);

  });
});
