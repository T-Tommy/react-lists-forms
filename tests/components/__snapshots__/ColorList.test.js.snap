// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList component renders ColorList 1`] = `
<Fragment>
  <CreateColor
    addColor={[Function]}
  />
  <Colors
    colors={Array []}
  />
</Fragment>
`;
